core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -60,14 +68,32 @@ private enum State {
     /** The response being decoded currently. */
     @Nullable
     private HttpResponseWrapper res;
-    @Nullable
-    private KeepAliveHandler keepAliveHandler;
+    private final KeepAliveHandler keepAliveHandler;
     private int resId = 1;
     private int lastPingReqId = -1;
     private State state = State.NEED_HEADERS;
 
-    Http1ResponseDecoder(Channel channel) {
+    Http1ResponseDecoder(Channel channel, HttpClientFactory clientFactory, SessionProtocol protocol) {
         super(channel, InboundTrafficController.ofHttp1(channel));
+        final long idleTimeoutMillis = clientFactory.idleTimeoutMillis();
```

Comment:
Moved the logic for creating `keepAliveHandler` into this constructor and `keepAliveHandler` is not nullable anymore.